File: Homework4/Calculator.fs
module Homework4.Calculator
let Calculate (val1: int) (operation: CalculatorOperation) (val2: int) =
    match operation with
    | CalculatorOperation.Plus -> val1 + val2
    | CalculatorOperation.Multiply -> val1 * val2
    | CalculatorOperation.Minus -> val1 - val2
    | _ -> val1 / val2

File: Homewrok8/Calculator/Calculator.cs
using System.Globalization;

namespace Homewrok8.Calculator
{
    public class Calculator : ICalculator
    {
        public string Add(double val1, double val2)
        {
            return (val1 + val2).ToString(CultureInfo.InvariantCulture);
        }

        public string Multiply(double val1, double val2)
        {
            return (val1 * val2).ToString(CultureInfo.InvariantCulture);
        }

        public string Divide(double val1, double val2)
        {
            return val2 == 0 ? "Infinity" : (val1 / val2).ToString(CultureInfo.InvariantCulture);
        }

        public string Minus(double val1, double val2)
        {
            return (val1 - val2).ToString(CultureInfo.InvariantCulture); 
        }
    }
}